lib/generator/print_helpers.dart

```diff
@@ -286,6 +292,25 @@ Spec generateLibrarySpec(LibraryDefinition definition) {
     );
   }
 
+  var importMeta = false;
+
+  // inserts import of meta package only if there is at least one non nullable input
+  // see this link for details https://github.com/dart-lang/sdk/issues/4188#issuecomment-240322222
+  definition.queries.forEach((definition) {
```

Comment:
I'm thinking about extracting this to a function, so you could just short-circuit the fn when a `input.isNonNull` is found, return true so you could import the directive. Also, `importMeta` var wouldn't be needed anymore.